client/java/src/main/java/com/linecorp/centraldogma/internal/client/AbstractWatcher.java

```diff
@@ -230,18 +235,17 @@ private void doWatch(int numAttemptsSoFar) {
         }
 
         final Revision lastKnownRevision = latest != null ? latest.revision() : Revision.INIT;
-        final CompletableFuture<Latest<T>> f = doWatch(client, projectName, repositoryName, lastKnownRevision);
+        final CompletableFuture<Latest<T>> f = doWatch(client, projectName, repositoryName, lastKnownRevision, timeoutMillis, errorOnEntryNotFound);
```

Comment:
How about specifying `errorOnEntryNotFound` as `true` (when it's true) for the initial attempt only?
```
private void doWatch(int numAttemptsSoFar, boolean initialAttempt) {
    ...
    doWatch(client, projectName, repositoryName, lastKnownRevision, timeoutMillis,
            initialAttempt? errorOnEntryNotFound : false);
    ...
}
```
so that the `Watcher` does not throw an exception when sending another watch request after an entry is removed?

- Send a watch request /foo.json which exists currently.
- Get notified when the /foo.json is removed.
- If the watcher sends another watch request to watch the creation of /foo.json, it will get an exception.

Reply:
I think your intention is good.
But I'm worried `initialValueFuture` & `initialAttempt` have a similar property. 
(I have a question about that `initialAttempt` is really required?)
So I'm going to try `!initialValueFuture.isDone() ? errorOnEntryNotFound : false`.
Please let me know if you have any concerns.

Reply:
> So I'm going to try !initialValueFuture.isDone() ? errorOnEntryNotFound : false.

That's way better. 😄